2021q3/current-compilation-time-analyze.adoc

```diff
@@ -0,0 +1,91 @@
+==  -CURRENT compilation time analyze
+
+Links: +
+
+link:https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=257141[Bug 257141] +
+link:https://lists.freebsd.org/archives/freebsd-current/2021-September/index.html#msg511[Discussion on freebsd-current]
+
+Contact: Wolfram Schneider <wosch@FreeBSD.org>
+
+Re-building FreeBSD from source takes a lot of CPU resources. Depending on your machine it takes between 20min and several hours. At the end of `make buildworld' we log the real time and which parameters are used for parallel build. E.g.
+
+--------------------------------------------------------------
+time make -j $(sysctl -n hw.ncpu) buildworld > buildworld.log 2>&1
+tail buildworld.log
+>>> World build completed on Sat Sep  4 20:58:00 UTC 2021
+>>> World built in 7235 seconds, ncpu: 3, make -j3
+     7235.61 real     20527.30 user       915.88 sys
+--------------------------------------------------------------
+
+
+The build process runs in several steps. It would be great to know which step takes most of the time, and what are the effects of special build parameters. With a small patch in Aug 2021 we get this information now:
+
+
+--------------------------------------------------------------
+egrep '>>> stage| real ' buildworld.log
+>>> stage 1.1: legacy release compatibility shims
+        0.28 real         0.18 user         0.10 sys
+>>> stage 1.2: bootstrap tools
+      165.99 real       472.58 user        11.56 sys
+>>> stage 2.1: cleaning up the object tree
+       21.47 real        36.96 user        14.14 sys
+       15.87 real        29.14 user        11.87 sys
+>>> stage 2.3: build tools
+        2.42 real         3.79 user         0.62 sys
+>>> stage 3: cross tools
+        9.92 real        18.49 user         1.75 sys
+>>> stage 3.1: recording build metadata
+        0.07 real         0.01 user         0.06 sys
+>>> stage 4.1: building includes
+       16.62 real        36.46 user         9.48 sys
+>>> stage 4.2: building libraries
+     5440.89 real     15724.60 user       482.58 sys
+>>> stage 4.3: building lib32 shim libraries
+      615.91 real      1654.77 user       164.58 sys
+>>> stage 4.4: building everything
+      937.23 real      2540.06 user       205.47 sys
+--------------------------------------------------------------
+
+In this example, we spent most of the time in "stage 4.2: building libraries", 77% of the CPU time and 75% of the real time.
+Now running the buildworld with the parameter "WITHOUT_TOOLCHAIN=yes" we get a much 3.3x faster build. Instead 2h it will be done in 36 minutes!
```

Comment:
s/much 3.3x faster/3.3x faster/ unless you prefer "much faster".

s/Instead/Instead of/